edx_exams/apps/core/tests/test_api.py

```diff
@@ -298,6 +299,33 @@ def test_submit_attempt(self):
             self.assertEqual(updated_attempt.status, ExamAttemptStatus.submitted)
             self.assertEqual(updated_attempt.end_time, timezone.now())
 
+    @patch('edx_exams.apps.core.signals.signals.EXAM_ATTEMPT_SUBMITTED.send_event')
```

Comment:
Should this be in a separate file for testing signals specifically?

Reply:
The signal is emitted from the API, so I think that this test make the most sense in this file, because that's what I'm testing. I could move it into a separate class specifically for testing events emitted by the API functions, if you'd prefer?

If I were to have tests in `signals.py`, they would be to test that calling `emit_exam_attempt_submitted_event` results in an emitted signal. Do you think that's a useful test to have? I could add that.

Reply:
Ah I see what you're saying, yeah it makes sense to do that test on the API level.

I guess I'm moreso trying to figure out if we need to do any testing on the level of the signal itself. I'm doing some digging now to see if that's necessary.

Reply:
Let me know what you find! I don't think we need to test the actual Django signal API. All the plumbing that funnels Django signals to the event bus is all behind the scenes and not something we could test, I believe. But we could test whether calling `emit_exam_attempt_submitted_event` results in an emitted signal.

Reply:
Ok I found an example, there's a [test here](https://github.com/openedx/edx-platform/blob/e45460a2ce6777c6440a76c5fbb2181d5edfa436/docs/hooks/events.rst#L4) in edx-platform certificates that seems similar to yours.

That tests seems to go about as deep as yours does so I think it's good as is.